src/k8s/pkg/utils/k8s/node_config.go

```diff
@@ -0,0 +1,57 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/mitchellh/mapstructure"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	applyv1 "k8s.io/client-go/applyconfigurations/core/v1"
+)
+
+func (c *Client) WatchNodeConfig(ctx context.Context, snap snap.Snap, reconcile func(ctx context.Context, snap snap.Snap, config types.NodeConfig) error, cancel func()) {
```

Comment:
Can we just make this a `WatchResource(namespace, name)`?

Reply:
We can do that with generics and pushing the data handling(by passing the whole resource) down the line, I kind of followed the spec. I don't have a strong preference, would like to hear from @neoaggelos on this as well.

Reply:
The kubernetes client should use kubernetes terms. So, let's make this a `WatchConfigMap(ctx, namespace, name, reconcile func(o *corev1.ConfigMap) error)`

Note that the reconcile function only gets the configmap object, everything else would come as a closure from the caller site